### ISSUE TITLE
Why findsecbugs report the vulnerabilities(Type: Hard coded password) in the code segment of the imported library and Class name definition

### ISSUE THREAD

**Issue by j0ck66** [find-sec-bugs/find-sec-bugs#617]
Example 1: Report the vulnerability in the code segment of the imported library

![image](https://user-images.githubusercontent.com/46952968/99230628-8a5cb100-282a-11eb-8001-15bec862bf50.png)

Example 2: The vulnerability in the line of class name definition.

![image](https://user-images.githubusercontent.com/46952968/99230395-3d78da80-282a-11eb-8d21-4d9f7a0ee6ae.png)

And the strange thing is that even setting the vulnerability as a false positive does not work, because the vulnerability will be reported again once it is rescanned.
What should I do to solve the problem...

**Comment by h3xstream** [comment]
I am seeing the package is `com.iasp...` which is propably private code. I won't be able to give a precise diagnostic.

What is most likely triggering this bug is the [ConstantPasswordDetector.java](https://github.com/find-sec-bugs/find-sec-bugs/blob/master/findsecbugs-plugin/src/main/java/com/h3xstream/findsecbugs/password/ConstantPasswordDetector.java). There are many detector for hardcoded password. This one is the only one reporting field containing potentially hardcoded key or password.
It determine potential keys by seeing field or local variable being pass to cryptographic function.

**TLDR: Look at the field of potential keys, IV or passwords.**

**Comment by j0ck66** [comment]
It is HardcodedPasswordEqualsDetector.java and IntuitiveHardcodePasswordDetector.java that trigger this vulnerability.
When I added related printing in the IntuitiveHardcodePasswordDetector.java and AbstractInjectionDetector.java, I found the following situation.
![image](https://user-images.githubusercontent.com/46952968/100322782-bc76cb80-2fff-11eb-8115-652c91cd9fce.png)
![image](https://user-images.githubusercontent.com/46952968/100323044-1c6d7200-3000-11eb-9d89-cd728ec4612b.png)
And the shown below is my log
![image](https://user-images.githubusercontent.com/46952968/100323312-7d954580-3000-11eb-819e-04655eafa2ef.png)

The module of report shows a problem on line 13 of the file，but the line content is like this 'import xx.yy.zz',and there is no content like 'setTradePassword'.

**Comment by j0ck66** [comment]
The version I use is 1.10.1

**Comment by j0ck66** [comment]
> I am seeing the package is `com.iasp...` which is propably private code. I won't be able to give a precise diagnostic.
> 
> What is most likely triggering this bug is the [ConstantPasswordDetector.java](https://github.com/find-sec-bugs/find-sec-bugs/blob/master/findsecbugs-plugin/src/main/java/com/h3xstream/findsecbugs/password/ConstantPasswordDetector.java). There are many detector for hardcoded password. This one is the only one reporting field containing potentially hardcoded key or password.
> It determine potential keys by seeing field or local variable being pass to cryptographic function.
> 
> **TLDR: Look at the fields of the class `CommonUtils` for potential keys, IV or passwords.**
> 
> I'm not sure about the first example. It could incorrect mapping of line numbers by the SpotBugs Sonar plugin.

This problem is caused by the automatic code generation of annotations.
When I canceled the annotation function,the line numbers was displayed normally.

**Comment by h3xstream** [comment]
So it was cause by bytecode generated by Aspect Oriented Programming (something like AspectJ) ?

**Comment by h3xstream** [comment]
Reopen because of #665

**Comment by h3xstream** [comment]
I need code or a sample project to reproduce the case.

What is needed : 
 - Class that contains `setTradePassword` (First)
 - Bytecode generate (If needed)

If you can isolate the code that is important in a separate class to avoid spoiling IP, this would be enough for me to investigate.

**Comment by h3xstream** [comment]
Without sample this issue can't be reproduce efficiently.

The bytecode and original are need. The `setTradePassword()` method called with a constant value. This would explain the bug being triggered.
Now, why is it not mapped to the proper line of code is something I don't have an explanation for. Looking at the input code and the actual execution of FSB would highlight the issue directly.
